File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    // using hashmaps
    let map = {}
    for (let i = 0; i < nums.length; i++) {
        const find = target - nums[i];

        if (map.hasOwnProperty(find)) {
            return [map[find], i]
        }
        
        map[nums[i]] = i
    }
};

File: 1-two-sum/two-sum.go
func twoSum(nums []int, target int) []int {
    indexMap := make(map[int]int)

    for i, value:= range nums {
        find := target - value;
        if index, ok := indexMap[find]; ok {
            return []int{index, i}
        }

        indexMap[value] = i
    }
    return nil
}